addons/fullscreen-editor/userscript.js

```diff
@@ -0,0 +1,61 @@
+import { setStageSize, setup } from "../mouse-pos/stage-size.js";
+
+/**
+ *
+ * @param {import("../../addon-api/content-script/typedef.js").UserscriptUtilities} param0
+ */
+export default async function ({ addon, console, msg }) {
+  setup(addon.tab.redux);
+  const fullscreen = document.createElement("a");
+  fullscreen.title = msg("fullscreen");
+  fullscreen.className = addon.tab.scratchClass("menu-bar_menu-bar-item", "menu-bar_hoverable", {
+    others: "sa-fullscreeneditor",
+  });
+
+  let isFullscreen = false;
+
+  const enterFullscreen = () => {
+    document.body.requestFullscreen();
+    if (addon.settings.get("shrinkStage")) {
+      if (!addon.settings.get("hide")) {
```

Comment:
It's possible for both settings to be enabled at the same time, even if the one to hide this stage is hidden from the settings page.